src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine_ProjectProcessing.cs

```diff
@@ -12,69 +12,9 @@
 namespace Microsoft.CodeAnalysis.FindSymbols
 {
     using DocumentMap = MultiDictionary<Document, (ISymbol symbol, IReferenceFinder finder)>;
-    using ProjectToDocumentMap = Dictionary<Project, MultiDictionary<Document, (ISymbol symbol, IReferenceFinder finder)>>;
 
     internal partial class FindReferencesSearchEngine
     {
-        private async Task ProcessProjectsAsync(
-            IEnumerable<ProjectId> connectedProjectSet,
-            ProjectToDocumentMap projectToDocumentMap)
-        {
-            var visitedProjects = new HashSet<ProjectId>();
-
-            // Make sure we process each project in the set.  Process each project in depth first
-            // order.  That way when we process a project, the compilations for all projects that it
-            // depends on will have been created already.
-            foreach (var projectId in connectedProjectSet)
-            {
-                _cancellationToken.ThrowIfCancellationRequested();
-
-                await ProcessProjectAsync(
-                    projectId, projectToDocumentMap, visitedProjects).ConfigureAwait(false);
-            }
-        }
-
-        private async Task ProcessProjectAsync(
-            ProjectId projectId,
-            ProjectToDocumentMap projectToDocumentMap,
-            HashSet<ProjectId> visitedProjects)
-        {
-            // Don't visit projects more than once.  
-            if (visitedProjects.Add(projectId))
-            {
-                var project = _solution.GetProject(projectId);
-
-                // Visit dependencies first.  That way the compilation for a project that we depend
-                // on is already ready for us when we need it.
-                foreach (var dependent in project.ProjectReferences)
-                {
-                    _cancellationToken.ThrowIfCancellationRequested();
-
-                    await ProcessProjectAsync(
-                        dependent.ProjectId, projectToDocumentMap, visitedProjects).ConfigureAwait(false);
-                }
-
-                await ProcessProjectAsync(project, projectToDocumentMap).ConfigureAwait(false);
-            }
-        }
-
-        private async Task ProcessProjectAsync(
-            Project project,
-            ProjectToDocumentMap projectToDocumentMap)
-        {
-            if (!projectToDocumentMap.TryGetValue(project, out var documentMap))
-            {
-                // No files in this project to process.  We can bail here.  We'll have cached our
-                // compilation if there are any projects left to process that depend on us.
-                return;
-            }
-
-            projectToDocumentMap.Remove(project);
-
-            // Now actually process the project.
-            await ProcessProjectAsync(project, documentMap).ConfigureAwait(false);
-        }
-
         private async Task ProcessProjectAsync(
```

Comment:
this is the only bit of code in this file, could probably just be moved to FindReferencesSearchEngine.cs now that it's much simpler

Reply:
Agreed.  can clean up in my continuing followup PRs :)